contracts/cw-storage/src/state.rs

```diff
@@ -1 +1,40 @@
+use crate::msg::BucketLimits;
+use cosmwasm_std::Uint128;
+use cw_storage_plus::Item;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+pub struct Bucket {
+    /// The name of the bucket.
+    pub name: String,
+    /// The limits of the bucket.
+    pub limits: Limits,
+}
+
+/// Limits is the type of the limits of a bucket.
+///
+/// The limits are optional and if not set, there is no limit.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+pub struct Limits {
+    /// The maximum total size of the objects in the bucket.
+    pub max_total_size: Option<Uint128>,
+    /// The maximum number of objects in the bucket.
+    pub max_objects: Option<Uint128>,
+    /// The maximum size of the objects in the bucket.
+    pub max_object_size: Option<Uint128>,
+    /// The maximum number of pins in the bucket for an object.
+    pub max_object_pins: Option<Uint128>,
+}
```

Comment:
I propose to use the `u128` primitive instead to ease its use, I think the `Uint128` shall be used only in the messages definition as its purpose is to provide string json serialization.

Reply:
Mmm 🤔, this one is also use to be saved in the state, so I'm wondering if it's not necessary to use the `Uint128` like it's done on other [cosmwasm smart contract](https://github.com/CosmWasm/cw-plus/blob/main/contracts/cw20-base/src/state.rs#L12) state to ensure good encoding into the state.

Reply:
@bdeneux Oh good point I think you're right! The CosmWasm storage use the json serialization so we need to keep the `Uint128`.